### ISSUE TITLE
Cannot connect to Eclipse-Che multiuser on Windows 2016 Hyper-V Ubuntu except from the VM

### ISSUE THREAD

**Issue by AMWild** [eclipse-che/che#10666]
### Description

I have a Windows Server 2016 machine with Hyper-V, on which I have a client Ubuntu OS operating.

I installed Eclipse Che multiuser to the Ubuntu VM.  Eclipse Che would appear to be working as expected when accessed from a browser on that machine, connecting initially to http://192.168.0.249:8080 (that being the IP address of the Ubuntu VM).

However... when I try to connect to Eclipse Che from the host Windows Server 2016 OS, using the same URL as the Ubuntu connection that succeeded, it redirects to http://192.168.0.249:8080/dashboard, and progresses no further, showing a black page.

When I attempt to connect to Eclipse Che from a separate Windows 10 machine on the same network - one that has been used successfully to connect to the WS2016's RDP, I gen no result at all, only an INET_E_RESOURCE_NOT_FOUND error.

I was not able to set up Eclipse Che to use my network's external address with port forwarding on my router - I got connectivity probems.

How can I connect to Eclipse Che on the Ubuntu VM from the host and other PCs on the network, and from internet PCs

### Reproduction Steps

1. Install Ubuntu Server 18.04

2. Install Ubuntu GUI

3. Share folder Eclipse-Che as NFS share on WS2016

4. Edit fstab on Ubuntu to connect folder /che-data to the NFS share.

5. Install Eclipse-Che. with docker ... -v /che-data:/data

6. Attempt to connect.  On failure, turn off Windows Server and Windows 10 OS' firewalls, with no effect.

7. Reset iptables in Ubuntu to:

-P INPUT ACCEPT
-P FORWARD ACCEPT
-P OUTPUT ACCEPT

... again, with no success.
 

**OS and version:**

Host: Windows Server 2016 Datacentre 
Client: Ubuntu Server 18.04

**Diagnostics:**

**Comment by AMWild** [comment]
@eivantsov I have CHE_HOST = 192.168.0.249, which is pingable from both the Win 2016 host machine and the machine that I would be doing the most work.

From my Win 10 laptop, all I get is a blank page, there is no indication that there has been any connection to Che at all:

DOM7011: The code on this page disabled back and forward caching. For more information, see: http://go.microsoft.com/fwlink/?LinkID=291337
www.google.com.au
HTML1300: Navigation occurred.
dnserror.htm
DOM7011: The code on this page disabled back and forward caching. For more information, see: http://go.microsoft.com/fwlink/?LinkID=291337
www.google.com.au
'bjlgo' is undefined


From my Win 2016 machine (which is hosting the VM that docker and Che reside upon), upon entering:

http://192.168.0.249:8080

it immediately navigates to http://192.168.0.249:8080/dashboard/ at which point everything stops.  Looking at the IE F12 stuff, I can see that there is an error:

DOM7011: The code on this page disabled back and forward caching. For more information, see: http://go.microsoft.com/fwlink/?LinkID=291337
about:tabs
HTML1300: Navigation occurred.
192.168.0.249:8080
DOM7011: The code on this page disabled back and forward caching. For more information, see: http://go.microsoft.com/fwlink/?LinkID=291337
about:tabs
SCRIPT5009: 'Promise' is undefined
app-e48d688d5a.js (13,1398)

If I add linebreaks in the .js file where my amateurish knowledge of JS suggests they ought to be, the indicated line is line 18:

function r(e){return new Promise(function(t,n){var i=document.createElement("script");i.async=!0,i.src=e["che.keycloak.js_adapter_url"],i.addEventListener("load",t),i.addEventListener("error",function(){return n("Error loading script.")}),i.addEventListener("abort",function(){return n("Script loading aborted.")}),document.head.appendChild(i)})}

The cursor was placed at Promise.

I have tried turning off the firewalls on both the Win10 and Win2016 machines without success, and the results are the same using either IE11 or Edge on Win10, or .

Additionally, I have a Synology rackstation on my network with Docker installed on it.  As it does not have enough RAM to run Che, I set up the ubuntu VM on my Win2016 server instead.  In case there was an IP conflict between my Synology box's docker and the ubuntu VM's docker, I changed the ubuntu VM's docker bridge IP address from 172.17.0.1 to 172.17.1.1 before installing Eclipse Che, but doing this does not seem to have made any difference - before or after the change, there was no connection from Win10, loading aborted at Dashboard for Win2016, and apparently normal behaviour from Firefox on the Ubuntu VM.

My Synology box _should_ play no part in Eclipse Che's installation - everything should be on the ubuntu VM on the Win2016 server, however, I must mention it as a potential source of other conflicts.

**Comment by skabashnyuk** [comment]
@AMWild Che is not certified to work on IE. I've heard about some chances to run on Edge(can't guarantee). I think the best results you can get on latest Chrome.

**Comment by AMWild** [comment]
@skabashnyuk I think that I've worked that out ;).

That said, it needs to be a lot clearer that IE will not work, preferably by coding Che so that if you open it with IE, it will tell you so in no uncertain terms, but also in the documentation.

I still can't account for why  Che would not do anything on my laptop for so long, through numerous reboots, on IE , Edge or Chrome, regardless of firewall settings or lack thereof.

I wasted a lot of time thinking that there was something wrong with my Win2016 server's settings because Che would not work with IE, and the documentation was not clear on that subject, and the code I looked at didn't do any detection of IE either.  Not that I'm blaming you... I should probably have ditched IE for good long ago already, and probably would have if I didn't have to deal with legacy systems on a regular basis that will _only_ work on IE.
